### PR TITLE
Fixed docs for the shape of `scores` in `generate()`

### PR DESCRIPTION
**Pull request by kylie-box**
# What does this PR do?
Fixed the shape of `scores` from `generate()` outputs to be `max_length-1` for all output classes. The first token `decoder_start_token_id` is not generated and thus no scores. 

[Scores in generate()](https://discuss.huggingface.co/t/scores-in-generate/3450)
[Generation Probabilities: How to compute probabilities of output scores for GPT2](https://discuss.huggingface.co/t/generation-probabilities-how-to-compute-probabilities-of-output-scores-for-gpt2/3175)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

@patrickvonplaten

### CODE REVIEW

File: src/transformers/generation_utils.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ class BeamSampleDecoderOnlyOutput(ModelOutput):
         scores (:obj:`tuple(torch.FloatTensor)` `optional`, returned when ``output_scores=True`` is passed or when ``config.output_scores=True``):
             Processed beam scores for each vocabulary token at each generation step. Beam scores consisting of log
             softmax scores for each vocabulary token and sum of log softmax of previously generated tokens in this beam
-            . :obj:`(max_length,)`-shaped tuple of :obj:`torch.FloatTensor` with each tensor of shape
+            . :obj:`(max_length-1,)`-shaped tuple of :obj:`torch.FloatTensor` with each tensor of shape
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:269]: For the "Decoder-only" outputs the length of scores should actually be equal to `max_length - input_sequence_length` whereas `input_sequence_length = input_ids.shape[-1]`. Could you update this maybe for all "DecoderOnly" outputs? :-)

Comment (kylie-box) [reply src/transformers/generation_utils.py]: I changed the `scores` from the "DecoderOnly" outputs to be `max_length - input_ids.shape[-1]`. Or is it better to change that to be exactly what you suggested (use `input_sequence_length`)?

**Comment by patrickvonplaten** [comment]
Hey @kylie-box, 

Could you run `make style` to fix a problem with the code quality. I think we can merge afterward :-)

**Comment by kylie-box** [comment]
Hey @patrickvonplaten,

It's fixed now. :)

**Comment by patrickvonplaten** [comment]
Thanks a lot!

**Pull request merged**